--- Fork.pm.orig	2014-08-15 12:10:51.155688251 +0400
+++ Fork.pm	2014-08-15 12:11:31.007688228 +0400
@@ -543,7 +543,7 @@
       require AnyEvent::Fork::Serve;
       $AnyEvent::Fork::Serve::OWNER = $parent;
       close $fh;
-      $0 = "$_[1] of $parent";
+      $0 = "$_[1] of $parent" if ($0 ne '-e');
       AnyEvent::Fork::Serve::serve ($slave);
       exit 0;
    } elsif (!$pid) {
--- Fork/Serve.pm.orig	2014-08-15 12:11:48.007688218 +0400
+++ Fork/Serve.pm	2014-08-15 12:12:29.943688194 +0400
@@ -55,7 +55,7 @@
          my $pid = fork;
 
          if ($pid eq 0) {
-            $0 = "AnyEvent::Fork of $OWNER";
+            $0 = "AnyEvent::Fork of $OWNER" if ($0 ne '-e');
             $master = pop @arg;
 
          } else {
@@ -77,7 +77,7 @@
          # we could free &serve etc., but this might just unshare
          # memory that could be shared otherwise.
          @_ = ($master, @arg);
-         $0 = "$buf of $OWNER";
+         $0 = "$buf of $OWNER" if ($0 ne '-e');
          package main;
          goto &$buf;
 
