--- Pool.pm.orig	2014-08-15 12:17:42.551688017 +0400
+++ Pool.pm	2014-08-15 12:19:47.231687946 +0400
@@ -481,19 +481,26 @@
 BEGIN {
    if ($^O eq "linux") {
       *ncpu = sub(;$) {
-         my ($cpus, $eus);
+         my ($cpus, $eus) = (0, 0);
 
          if (open my $fh, "<", "/proc/cpuinfo") {
             my %id;
+            my $unit;
 
             while (<$fh>) {
-               if (/^core id\s*:\s*(\d+)/) {
-                  ++$eus;
-                  undef $id{$1};
-               }
+                if (/^physical id\s*:\s*(\d+)/) {
+                    $unit = $1;
+                }
+                elsif (/^core id\s*:\s*(\d+)/) {
+                    defined $unit && $id{$unit}++;
+                    undef $unit;
+                }
             }
 
             $cpus = scalar keys %id;
+            $eus += $_ for (values %id);
+
+            close $fh;
          } else {
             $cpus = $eus = @_ ? shift : 1;
          }
