--- Pool.pm	2013-04-28 18:19:07.000000000 +0400
+++ /home/tvv/.staticperl-5.18.1/lib/AnyEvent/Fork/Pool.pm	2013-10-25 12:25:23.101617590 +0400
@@ -481,19 +481,25 @@
 BEGIN {
    if ($^O eq "linux") {
       *ncpu = sub(;$) {
-         my ($cpus, $eus);
+         my ($cpus, $eus) = (0, 0);
 
          if (open my $fh, "<", "/proc/cpuinfo") {
             my %id;
+            my $unit;
 
             while (<$fh>) {
-               if (/^core id\s*:\s*(\d+)/) {
-                  ++$eus;
-                  undef $id{$1};
-               }
+                if (/^physical id\s*:\s*(\d+)/) {
+                    $unit = $1;
+                } elsif (/^core id\s*:\s*(\d+)/) {
+                    defined $unit && $id{$unit}++;
+                    undef $unit;
+                }
             }
 
             $cpus = scalar keys %id;
+            $eus += $_ for (values %id);
+
+            close $fh;
          } else {
             $cpus = $eus = @_ ? shift : 1;
          }
