--- Makefile.PL.orig    2013-12-23 10:50:43.353559493 +0400
+++ Makefile.PL 2013-12-23 10:51:09.983826393 +0400
@@ -406,7 +406,8 @@
     'LIBS'           => join(" ", @libs),
     'clean'         => {'FILES' => 'xssubs.c conf Tty.exp_old log'},
     'realclean'             => {'FILES' => 'Tty IO-Tty.ppd'},
-    'MAP_TARGET'     => 'perltty',
+# staticperl expects MAP_TARGET = perl
+#    'MAP_TARGET'     => 'perltty',
     'AUTHOR'         => 'Roland Giersig <RGiersig@cpan.org>',
     'ABSTRACT'       => 'Pseudo ttys and constants',
     'LICENSE'        => 'perl',

